disco_aws_automation/socify_helper.py

```diff
@@ -0,0 +1,96 @@
+        self._ticket_id = ticket_id
+        self._command = command
+        self._sub_command = sub_command
+
+        if config:
+            self._config = config
+        else:
+            self._config = read_config()
+        self._socify_url = None
+
+    def _build_event_url(self):
+        """
+        Build the socify event url using the socify configuration data provided in the disco_aws.ini file
+        :return: The socify URL
+        """
+        if not self._socify_url:
+            try:
+                self._socify_url = self._config.get("socify", "socify_baseurl")
+            except NoOptionError:
+                logger.warning("The property socify_baseurl is not set in your disco_aws.ini file. The "
+                               "deploy action won't be logged in your ticket. Please make sure to add the "
+                               "definition for socify_baseurl in the [socify] section.")
+                return
+        return self._socify_url + "/event"
+
+    def _build_event_json(self, status, **kwargs):
+        """
+        Build the event JSON for the Socify Event associated to the executed command
+        :param status: The status of the executed command that we are going to log
+        :param hostclass: The hostclass for which the command was executed
+        :param message: An optional error message
+        :return: The Event JSON for the associated Event
+        """
+        event_info = {'status': status}
+        if self._sub_command:
+            event_info['sub_cmd'] = self._sub_command
+
+        event_info.update(kwargs)
+
+        event_json = {"ticketId": self._ticket_id,
+                      "cmd": self._command,
+                      "data": event_info}
+
+        return event_json
+
+    def send_event(self, status, **kwargs):
+        """
+        helper function used to send a socify event
+        :param status: The status of the executed command that we are going to log
+        :param hostclass: The hostclass for which the command was executed
+        :param message:An optional error message
+        """
+        if not self._ticket_id:
+            return
+
+        url = self._build_event_url()
+        if not url:
+            # We were not able to build the event URL
+            return
+
+        data = self._build_event_json(status, **kwargs)
+        try:
+            headers = {'Content-Type': 'application/json'}
+            response = requests.post(url=url, headers=headers, json=data)
+            response.raise_for_status()
+            status = response.status_code
+            rsp_json = response.json()
+            logger.info("received response status %s data: %s", status, rsp_json)
+        except requests.HTTPError:
+            rsp_json = response.json()
+            logger.error("Socify event failed with the following error: %s", rsp_json)
+        except Exception as err:
+            logger.error("Failed to send event to Socify: %s", err)
```

Comment:
You can do `logger.exception` and it will automatically include stacktrace in log